File: app/src/main/java/net/tlalka/android/fiszki/view/activities/TestListActivity.java
package net.tlalka.android.fiszki.view.activities;

import android.os.Bundle;
import android.view.View;
import android.widget.AdapterView;
import android.widget.ListView;
import butterknife.BindView;
import net.tlalka.android.fiszki.R;
import net.tlalka.android.fiszki.domain.services.LessonListService;
import net.tlalka.android.fiszki.model.dto.LessonDto;
import net.tlalka.android.fiszki.model.entities.Lesson;
import net.tlalka.android.fiszki.view.adapters.TestsAdapter;
import net.tlalka.android.fiszki.view.navigations.Navigator;

import javax.inject.Inject;

public class TestListActivity extends BasePageActivity implements AdapterView.OnItemClickListener {

    @BindView(R.id.test_list_view)
    protected ListView listView;

    @Inject
    protected LessonListService lessonListService;

    @Inject
    protected Navigator navigator;

    @Override
    public void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        super.setContentView(R.layout.test_list_activity);
        super.getActivityComponent().inject(this);

        this.initLessonsList();
    }

    private void initLessonsList() {
        listView.setAdapter(new TestsAdapter(this, lessonListService.getLessons()));
        listView.setOnItemClickListener(this);
    }

    @Override
    public void onItemClick(AdapterView<?> parent, View view, int position, long id) {
        Lesson lesson = (Lesson) parent.getItemAtPosition(position);

        this.navigator.openTestActivity(this, new LessonDto(lesson));
        this.navigator.finish(this);
    }
}

File: app/src/main/java/net/tlalka/android/fiszki/view/activities/TestListActivity.kt
package net.tlalka.android.fiszki.view.activities

import android.os.Bundle
import android.view.View
import android.widget.AdapterView
import android.widget.ListView
import butterknife.BindView
import net.tlalka.android.fiszki.R
import net.tlalka.android.fiszki.domain.controllers.ListController
import net.tlalka.android.fiszki.model.dto.LessonDto
import net.tlalka.android.fiszki.model.entities.Lesson
import net.tlalka.android.fiszki.view.adapters.TestsAdapter
import net.tlalka.android.fiszki.view.navigations.Navigator
import javax.inject.Inject

class TestListActivity : BasePageActivity(), AdapterView.OnItemClickListener {

    @BindView(R.id.test_list_view)
    lateinit var listView: ListView

    @Inject
    lateinit var listController: ListController

    @Inject
    lateinit var navigator: Navigator

    public override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        super.setContentView(R.layout.test_list_activity)
        super.activityComponent.inject(this)

        initLessonsList()
    }

    private fun initLessonsList() {
        listView.adapter = TestsAdapter(this, listController.getLessonList())
        listView.onItemClickListener = this
    }

    override fun onItemClick(parent: AdapterView<*>, view: View, position: Int, id: Long) {
        val lesson = parent.getItemAtPosition(position) as Lesson

        navigator.openTestActivity(this, LessonDto(lesson, position + 1))
        navigator.finish(this)
    }
}
